### ISSUE TITLE
Docker Desktop >=4.10.0 can't start Docker Engine on Ventura 13.3.1 (M2)

### ISSUE THREAD

**Issue by FirefoxMetzger** [docker/for-mac#6829]
- [x] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID (Docker Desktop v4.19.0): 3E6D0AF4-1336-46C7-892A-0B7D9A33B762/20230508124814
  - Diagnostics ID (Docker Desktop v4.10.1): 3E6D0AF4-1336-46C7-892A-0B7D9A33B762/20230508130441

**Note:** For reference, here are the diagnostics for v4.9.1 (which is the most recent working version I found):

Diagnostics ID (v4.9.2): 3E6D0AF4-1336-46C7-892A-0B7D9A33B762/20230508131331

### Expected behavior

After Installation Docker Desktop starts and I can execute commands using the docker CLI.

### Actual behavior

Docker Desktop doesn't manage to launch the Docker Engine. Instead it seems to be "stuck" on this step. Commands sent via the command line fail due to the engine not running and the Docker Desktop Dashboard is unresponsive with the exception of the Troubleshooting window.

It is also impossible to close Docker Desktop. (I have to force quit.)

### Information

  - macOS Version: Ventura 13.3.1
  - Intel chip or Apple chip: Apple Chip (M2)
  - Docker Desktop Version: 4.19.0

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

<details>

```
% /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check           
[2023-05-08T10:14:15.626612000Z][com.docker.diagnose][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[FAIL] DD0004: is the Docker engine running? Get "http://ipc/docker": dial unix lifecycle-server.sock: connect: no such file or directory
[2023-05-08T10:14:15.661035000Z][com.docker.diagnose][I] ipc.NewClient: 9798da73-com.docker.diagnose -> lifecycle-server.sock VMDockerdAPI
[2023-05-08T10:14:15.661182000Z][com.docker.diagnose][I] (977d328f) 9798da73-com.docker.diagnose C->S VMDockerdAPI GET /docker
[2023-05-08T10:14:15.661705000Z][com.docker.diagnose][W] (977d328f) 9798da73-com.docker.diagnose C<-S NoResponse GET /docker (517.5µs): Get "http://ipc/docker": dial unix lifecycle-server.sock: connect: no such file or directory
[2023-05-08T10:14:15.661899000Z][com.docker.diagnose][I] (977d328f-1) 9798da73-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-08T10:14:15.662188000Z][com.docker.diagnose][W] (977d328f-1) 9798da73-com.docker.diagnose C<-S NoResponse GET /ping (285.709µs): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2023-05-08T10:14:16.662646000Z][com.docker.diagnose][I] (977d328f-2) 9798da73-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-08T10:14:16.664243000Z][com.docker.diagnose][W] (977d328f-2) 9798da73-com.docker.diagnose C<-S NoResponse GET /ping (1.574ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2023-05-08T10:14:17.665511000Z][com.docker.diagnose][I] (977d328f-3) 9798da73-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-08T10:14:17.667489000Z][com.docker.diagnose][W] (977d328f-3) 9798da73-com.docker.diagnose C<-S NoResponse GET /ping (1.966834ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2023-05-08T10:14:18.668803000Z][com.docker.diagnose][I] (977d328f-4) 9798da73-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-08T10:14:18.670504000Z][com.docker.diagnose][W] (977d328f-4) 9798da73-com.docker.diagnose C<-S NoResponse GET /ping (1.69825ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2023-05-08T10:14:19.671817000Z][com.docker.diagnose][I] (977d328f-5) 9798da73-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-08T10:14:19.673866000Z][com.docker.diagnose][W] (977d328f-5) 9798da73-com.docker.diagnose C<-S NoResponse GET /ping (2.044125ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2023-05-08T10:14:20.675343000Z][com.docker.diagnose][I] (977d328f-6) 9798da73-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-08T10:14:20.677293000Z][com.docker.diagnose][W] (977d328f-6) 9798da73-com.docker.diagnose C<-S NoResponse GET /ping (1.937959ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2023-05-08T10:14:21.678552000Z][com.docker.diagnose][I] (977d328f-7) 9798da73-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-08T10:14:21.680234000Z][com.docker.diagnose][W] (977d328f-7) 9798da73-com.docker.diagnose C<-S NoResponse GET /ping (1.680458ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2023-05-08T10:14:22.681078000Z][com.docker.diagnose][I] (977d328f-8) 9798da73-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-08T10:14:22.682645000Z][com.docker.diagnose][W] (977d328f-8) 9798da73-com.docker.diagnose C<-S NoResponse GET /ping (1.559167ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory

[PASS] DD0015: are the binary symlinks installed?
[FAIL] DD0031: does the Docker API work? Get "http://docker.raw.sock/v1.24/containers/json": context deadline exceeded
[PASS] DD0013: is the $PATH ok?
[FAIL] DD0003: is the Docker CLI working? signal: killed
[PASS] DD0038: is the connection to Docker working?
[FAIL] DD0014: are the backend processes running? 1 error occurred:
	* com.docker.vpnkit is not running


[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[FAIL] DD0009: is the vpnkit API responding? dial unix vpnkit.diag.sock: connect: connection refused
[PASS] DD0010: is the Docker API proxy responding?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[WARN] DD0004: is the Docker engine running? Get "http://ipc/docker": dial unix lifecycle-server.sock: connect: no such file or directory
[PASS] DD0015: are the binary symlinks installed?
[WARN] DD0031: does the Docker API work? Get "http://docker.raw.sock/v1.24/containers/json": context deadline exceeded
[WARN] DD0032: do Docker networks overlap with host IPs? Get "http://docker.raw.sock/v1.24/networks": context deadline exceeded

Please note the following 3 warnings:

1 : The check: is the Docker engine running?
    Produced the following warning: Get "http://ipc/docker": dial unix lifecycle-server.sock: connect: no such file or directory

The Docker engine manages all containers and images on the host. Check the dockerd.log to see why it failed to start.

2 : The check: does the Docker API work?
    Produced the following warning: Get "http://docker.raw.sock/v1.24/containers/json": context deadline exceeded

If the Docker API is not available from the host then Docker Desktop will not work correctly.

3 : The check: do Docker networks overlap with host IPs?
    Produced the following warning: Get "http://docker.raw.sock/v1.24/networks": context deadline exceeded

If the subnet used by a Docker network overlaps with an IP used by the host, then containers
won't be able to contact the overlapping IP addresses.

Try configuring the IP address range used by networks: in your docker-compose.yml.
See https://docs.docker.com/compose/compose-file/compose-file-v2/#ipv4_address-ipv6_address


Please investigate the following 2 issues:

1 : The test: is the Docker engine running?
    Failed with: Get "http://ipc/docker": dial unix lifecycle-server.sock: connect: no such file or directory

The Docker engine manages all containers and images on the host. Check the dockerd.log to see why it failed to start.

2 : The test: are the backend processes running?
    Failed with: 1 error occurred:
	* com.docker.vpnkit is not running



Not all of the backend processes are running.
```

</details>

### Steps to reproduce the behavior

1. Follow the instructions found at https://docs.docker.com/desktop/install/mac-install/

**Comment by rcontreras-te** [comment]
Seeing the same thing.  After upgrading Docker Desktop doesn't want to start.

macOS: Ventura 13.3.1 (a) - M1
Docker Desktop: 4.18.0 (104112)

**Comment by cougarsyankeesfan** [comment]
Completely uninstalling Docker Desktop, rebooting, and reinstalling it on my Mac M1 fixed it for me. https://docs.docker.com/desktop/uninstall/

**Comment by amoshimei** [comment]
I am in the same situation.
Docker Desktop fails to start on MacOS Ventura 13.3.1 and 12 with M1Max chip
I am experiencing a persisting issue with Docker Desktop.

Observed behavior:
Upon launching Docker Desktop, the message 'Docker Desktop is starting' persists indefinitely.

System Information:
MacOS Version: Ventura 13.3.1 and 12
Processor Type: Apple M1MAX chip
Docker Desktop Version: Issues observed with 4.19.0, 4.12.0, 4.10.1; Version 4.9.1 functions correctly

Steps Attempted:
・Switching Docker Desktop versions:

・Moved Docker from the Applications folder to Trash
Downloaded and tested various versions from the Docker release notes (https://docs.docker.com/desktop/release-notes/#docker-desktop-481)

・Updated MacOS from version 12 to the latest version 13.

・Reinstalled Docker Desktop:
Clicked [Troubleshoot] -> [Uninstall] in Docker Desktop.
Updated Golang version to 1.20.3
Installed the latest Docker Desktop version from the official Docker installation guide (https://docs.docker.com/desktop/install/mac-install/)

All of these steps resulted in the 'Docker Desktop is starting' message persisting indefinitely when starting Docker Desktop.

Temporary Solution:
Currently, I am able to use Docker Desktop version 4.9.1 without this issue.

Expected behavior:
I hope to be able to utilize the latest version of Docker for Mac without encountering this startup issue.

Any assistance would be greatly appreciated. Thank you.

**Comment by FirefoxMetzger** [comment]
Good news. Docker released [v4.22.1](https://docs.docker.com/desktop/release-notes/#4221) last week. This appears to solve the problem I had with earlier versions, and it now runs on my M2.

(I haven't tested it beyond running the hello-world container, but at least the issue of it not starting has been fixed.)
